### DISCUSSION TITLE
Element type is invalid: expected a string (for built-in components) or a class/function

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#249] [Look at what I learned!]
I was getting this error when I was tracking to have a navbar with basic routes and views. 

This error was happening bc I was telling the route to be an exact path, but that view was returning nothing. 
That's why the error was stating "Element type is invalid: expected a string (for built-in components) or a class/function".

Once I gave the components a function that returned my navbar, the error went away!

**Comment by drteresavasquez** [comment]
Ah. So the component you were passing to your route wasn't returning anything?

**Reply by BenJarrett** [reply]
Correct! I simply just trying to get my basic routes set up and working, first. 
It was saying "Well what the hell do you want me to do with this component here? There is nothing for me to return?" 

That why it was expecting a string or class/function to tell it what to do!

**Reply by drteresavasquez** [reply]
🤣 this is also why I return the h1 tags in every page I build first. It's happened to us all!
